includes/batch.inc

```diff
@@ -241,8 +245,18 @@ function islandora_paged_content_create_ocr_batch_operation(array $pages, array
   $page = islandora_paged_content_start_batch_operation($pages, $context);
   if ($page) {
     $page = islandora_object_load($page);
-    $context['results']['successful'][] = islandora_paged_content_page_derive_ocr_datastreams($page, $options);
-    islandora_paged_content_end_batch_operation($context, 1, array('Performed OCR on @successful of @total pages.'));
+    $ocr = islandora_paged_content_page_generate_ocr_datastreams($page, $options, FALSE);
+    foreach ($ocr as $dsid => $result) {
+      if ($result !== FALSE) {
+        $context['results']['files'][] = $result;
+        // OCR files will be required for concatenation.
+        if ($dsid == 'OCR') {
+          $context['results']['pages'][] = $result;
+        }
+        $context['results']['successful'][] = islandora_paged_content_update_datastream($page, $result, $dsid);
+      }
+    }
+    islandora_paged_content_end_batch_operation($context, 1, array('Derived @successful of @total OCR/HOCR datastreams.'));
```

Comment:
Seems to lead to non-sensical messages... These batch-wrapping functions don't like running with multiple operations in a batch?